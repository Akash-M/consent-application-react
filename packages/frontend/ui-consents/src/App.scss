@import "lib-components/src/scss/main";

main {
  align-items: flex-start;
  display: flex;
  flex-flow: column nowrap;
  height: 100vh;
  width: 100%;
}
