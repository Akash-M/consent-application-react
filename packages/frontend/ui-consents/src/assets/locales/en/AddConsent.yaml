header: Give consent
username: Name
email: Email Address
disclaimer: 'I agree to:'
addButton: Give consent
success: Consent added successfully
errors:
  username: Please enter a valid name
  email: Please enter a valid email id
  consents: Please ensure that you agree to at least one consent
  api: Error fetching consent list. Please try again in some time.
