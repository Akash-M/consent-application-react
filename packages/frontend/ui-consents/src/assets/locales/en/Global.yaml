headers:
  addConsent: Give consent
  listConsent: Collected consents
consents:
  newsletter: Receive newsletter
  ads: Be shown targeted ads
  statistics: Contribute to anonymous visit statistics
loader: Loading...
