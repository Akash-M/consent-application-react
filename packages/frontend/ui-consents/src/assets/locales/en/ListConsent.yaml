header: Collected consents
table:
  headers:
    username: Name
    email: Email Address
    consent: Consent given for
  pagination:
    previous: Previous Page
    next: Next Page
    perPage: Rows per page
    all: All
errors:
  api: Error fetching consent list. Please try again in some time.
