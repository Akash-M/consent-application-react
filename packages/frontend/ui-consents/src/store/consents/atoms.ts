import { atom } from 'recoil';

export const ConsentListState = atom<Consent.Detail[]>({
  key: 'ConsentListState',
  default: [],
});

export const ConsentListPaginatorState = atom<Consent.Paginator>({
  key: 'ConsentListPaginatorState',
  default: {
    currentPage: 1,
    perPage: 2,
  },
});
