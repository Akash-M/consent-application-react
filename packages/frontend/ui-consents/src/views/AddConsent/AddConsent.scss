.add-consent {
  display: flex;
  flex-direction: column;
  margin: 1rem;

  @include tablet {
    width: auto;
  }

  @include desktop {
    margin: 0;
    max-width: 1024px;
    width: auto;
  }

  &__user-info {
    display: flex;
    flex-direction: column;
    margin-bottom: 2rem;

    .text-field {
      height: 40px;
    }

    .text-field:first-child {
      margin-bottom: 1rem;
    }

    @include tablet {
      flex-direction: row;

      .text-field {
        min-width: 250px;
      }

      .text-field:first-child {
        margin-bottom: 0;
      }

      .text-field:last-child {
        margin-left: 1rem;
      }
    }
  }

  &__consents {
    display: flex;
    flex-direction: column;
    margin-bottom: 2rem;
  }

  > button {
    align-self: flex-start;
    width: fit-content;
  }
}
