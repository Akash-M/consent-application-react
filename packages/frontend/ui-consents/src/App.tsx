import React from 'react';
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';

import '$/App.scss';
import SiteHeader from '$/components/SiteHeader';
import Layout from '$/containers/Layout';

const App = (): JSX.Element => {
  return (
    <main>
      <SiteHeader />

      <Layout />

      <ToastContainer />
    </main>
  );
};

export default App;
