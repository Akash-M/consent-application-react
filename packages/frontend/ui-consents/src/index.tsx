import React, { Suspense } from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import { RecoilRoot } from 'recoil';

import setI18n from 'lib-utils/src/i18n';

import App from '$/App';

export function Root(): JSX.Element {
  return (
    <React.StrictMode>
      <RecoilRoot>
        <Router>
          <Suspense fallback={null}>
            <App />
          </Suspense>
        </Router>
      </RecoilRoot>
    </React.StrictMode>
  );
}

setI18n('Global', true, { en: '0.2' });

const rootElement = document.querySelector('#root');
ReactDOM.render(<Root />, rootElement);
