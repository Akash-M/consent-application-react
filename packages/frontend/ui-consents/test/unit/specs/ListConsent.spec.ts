import { fireEvent, screen, waitFor } from '@testing-library/react';
import { setI18n } from 'react-i18next';

import { consentListFixtures } from 'lib-api/src/fixtures';
import { customRenderer } from 'lib-utils/src/testing/factory';
import { I18N_MISSING_KEY, loadI18n } from 'lib-utils/src/testing/i18n';

import App from '$/App';
import AddConsent from '$/assets/locales/en/AddConsent.yaml';
import Global from '$/assets/locales/en/Global.yaml';
import ListConsent from '$/assets/locales/en/ListConsent.yaml';
import { ConsentListState } from '$/store/consents/atoms';

const initializeState = ({ set }: any) => {
  set(ConsentListState, consentListFixtures);
};

describe('<App />', () => {
  beforeAll(() => {
    setI18n(loadI18n('Global', { Global, AddConsent, ListConsent }));
  });

  beforeEach(jest.clearAllMocks);

  test('should render list of consents with 2 entries by default', async () => {
    const { container } = customRenderer(App, initializeState);
    fireEvent.click(screen.getByText('Global.headers.listConsent'));
    await waitFor(() => {
      expect(screen.getByText('user1@email.com')).toBeTruthy();
    });
    expect(container.firstChild!.textContent).not.toContain(I18N_MISSING_KEY);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('should change page when user clicks on paginator', async () => {
    const { container } = customRenderer(App, initializeState);
    fireEvent.click(screen.getByText('Global.headers.listConsent'));
    await waitFor(() => {
      expect(screen.getByText('user1@email.com')).toBeTruthy();
    });
    fireEvent.click(screen.getByLabelText(/next page/i));
    await waitFor(() => {
      expect(screen.getByText('user3@email.com')).toBeTruthy();
    });
    expect(container.firstChild).toMatchSnapshot();
  });
});
