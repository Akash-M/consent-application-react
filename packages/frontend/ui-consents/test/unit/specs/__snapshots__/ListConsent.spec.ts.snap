// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should change page when user clicks on paginator 1`] = `
<main>
  <nav
    class="site-header"
  >
    <h2>
      Consents App
    </h2>
  </nav>
  <article
    class="layout"
  >
    <aside
      class="layout__links"
    >
      <a
        class=""
        href="/"
      >
        Global.headers.addConsent
      </a>
      <a
        aria-current="page"
        class="active"
        href="/list-consent"
      >
        Global.headers.listConsent
      </a>
    </aside>
    <section
      class="layout__screens"
    >
      <article
        class="list-consent"
      >
        <h2>
          ListConsent.header
        </h2>
        <table
          class="ca-table"
        >
          <thead>
            <tr>
              <th>
                ListConsent.table.headers.username
              </th>
              <th>
                ListConsent.table.headers.email
              </th>
              <th>
                ListConsent.table.headers.consent
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              data-testid="consent-entry"
            >
              <td>
                user3
              </td>
              <td>
                user3@email.com
              </td>
              <td>
                <p />
                <p />
                <p />
              </td>
            </tr>
          </tbody>
        </table>
        <nav
          aria-label="pagination navigation"
          class="MuiPagination-root MuiPagination-outlined css-1oj2twp-MuiPagination-root"
        >
          <ul
            class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
          >
            <li>
              <button
                aria-label="Go to previous page"
                class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-circular MuiPaginationItem-previousNext css-lqq3n7-MuiButtonBase-root-MuiPaginationItem-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                  data-testid="NavigateBeforeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
            <li>
              <button
                aria-label="Go to page 1"
                class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-circular MuiPaginationItem-page css-lqq3n7-MuiButtonBase-root-MuiPaginationItem-root"
                tabindex="0"
                type="button"
              >
                1
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
            <li>
              <button
                aria-current="true"
                aria-label="page 2"
                class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-circular Mui-selected MuiPaginationItem-page css-lqq3n7-MuiButtonBase-root-MuiPaginationItem-root"
                tabindex="0"
                type="button"
              >
                2
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
            <li>
              <button
                aria-label="Go to next page"
                class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-lqq3n7-MuiButtonBase-root-MuiPaginationItem-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                  data-testid="NavigateNextIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
              </button>
            </li>
          </ul>
        </nav>
      </article>
    </section>
  </article>
  <div
    class="Toastify"
  />
</main>
`;

exports[`<App /> should render list of consents with 2 entries by default 1`] = `
<main>
  <nav
    class="site-header"
  >
    <h2>
      Consents App
    </h2>
  </nav>
  <article
    class="layout"
  >
    <aside
      class="layout__links"
    >
      <a
        class=""
        href="/"
      >
        Global.headers.addConsent
      </a>
      <a
        aria-current="page"
        class="active"
        href="/list-consent"
      >
        Global.headers.listConsent
      </a>
    </aside>
    <section
      class="layout__screens"
    >
      <article
        class="list-consent"
      >
        <h2>
          ListConsent.header
        </h2>
        <table
          class="ca-table"
        >
          <thead>
            <tr>
              <th>
                ListConsent.table.headers.username
              </th>
              <th>
                ListConsent.table.headers.email
              </th>
              <th>
                ListConsent.table.headers.consent
              </th>
            </tr>
          </thead>
          <tbody>
            <tr
              data-testid="consent-entry"
            >
              <td>
                user1
              </td>
              <td>
                user1@email.com
              </td>
              <td>
                <p>
                  Global.consents.newsletter
                </p>
                <p />
                <p>
                  Global.consents.statistics
                </p>
              </td>
            </tr>
            <tr
              data-testid="consent-entry"
            >
              <td>
                user2
              </td>
              <td>
                user2@email.com
              </td>
              <td>
                <p>
                  Global.consents.newsletter
                </p>
                <p>
                  Global.consents.ads
                </p>
                <p>
                  Global.consents.statistics
                </p>
              </td>
            </tr>
          </tbody>
        </table>
        <nav
          aria-label="pagination navigation"
          class="MuiPagination-root MuiPagination-outlined css-1oj2twp-MuiPagination-root"
        >
          <ul
            class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
          >
            <li>
              <button
                aria-label="Go to previous page"
                class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-lqq3n7-MuiButtonBase-root-MuiPaginationItem-root"
                disabled=""
                tabindex="-1"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                  data-testid="NavigateBeforeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                  />
                </svg>
              </button>
            </li>
            <li>
              <button
                aria-current="true"
                aria-label="page 1"
                class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-circular Mui-selected MuiPaginationItem-page css-lqq3n7-MuiButtonBase-root-MuiPaginationItem-root"
                tabindex="0"
                type="button"
              >
                1
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
            <li>
              <button
                aria-label="Go to page 2"
                class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-circular MuiPaginationItem-page css-lqq3n7-MuiButtonBase-root-MuiPaginationItem-root"
                tabindex="0"
                type="button"
              >
                2
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
            <li>
              <button
                aria-label="Go to next page"
                class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-circular MuiPaginationItem-previousNext css-lqq3n7-MuiButtonBase-root-MuiPaginationItem-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                  data-testid="NavigateNextIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </li>
          </ul>
        </nav>
      </article>
    </section>
  </article>
  <div
    class="Toastify"
  />
</main>
`;
